tiledb/sm/filesystem/vfs.cc

```diff
@@ -709,18 +679,18 @@ std::vector<directory_entry> VFS::ls_with_sizes(const URI& parent) const {
   return entries;
 }
 
-Status VFS::move_file(const URI& old_uri, const URI& new_uri) {
+void VFS::move_file(const URI& old_uri, const URI& new_uri) const {
   auto instrument = make_log_duration_instrument(old_uri, new_uri);
   // If new_uri exists, delete it or raise an error based on `force`
-  bool is_file;
-  RETURN_NOT_OK(this->is_file(new_uri, &is_file));
-  if (is_file)
-    RETURN_NOT_OK(remove_file(new_uri));
+  if (this->is_file(new_uri)) {
+    remove_file(new_uri);
+  }
```

Comment:
Same with `copy_file`. BTW we should strive towards removing as much logic from the `VFS` class as possible and pushing it down to the backends, or shared helpers.